src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/calendar/NextHearingDateIntervalData.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.calendar;
+
+import lombok.Builder;
+import lombok.Getter;
+
+import java.util.List;
+
+@Builder
+@Getter
+public class NextHearingDateIntervalData {
```

Comment:
Why do you need a new class for next hearing date data and and due date data. You can use the DateTypeIntervalData class for all the types